allennlp/data/dataset_readers/dataset_utils/span_utils.py

```diff
@@ -131,47 +129,48 @@ def bio_tags_to_spans(tag_sequence: List[str],
     return list(spans)
 
 
-class InvalidTagSequence(Exception):
-    pass
-
-
-def bioul_tags_to_spans(tag_sequence: List[str]) -> List[TypedStringSpan]:
+def bioul_tags_to_spans(tag_sequence: List[str],
+                        classes_to_ignore: List[str] = None) -> List[TypedStringSpan]:
     """
-    Given a sequence corresponding to BIOLU tags, extracts spans.
+    Given a sequence corresponding to BIOUL tags, extracts spans.
     Spans are inclusive and can be of zero length, representing a single word span.
     Ill-formed spans are not allowed and will raise ``InvalidTagSequence``.
 
     Parameters
     ----------
     tag_sequence : ``List[str]``, required.
-        The tag sequence encoded in BIOLU, e.g. ["B-PER", "L-PER", "O"].
+        The tag sequence encoded in BIOUL, e.g. ["B-PER", "L-PER", "O"].
+    classes_to_ignore : ``List[str]``, optional (default = None).
+        A list of string class labels `excluding` the bio tag
+        which should be ignored when extracting spans.
 
     Returns
     -------
     spans : ``List[TypedStringSpan]``
         The typed, extracted spans from the sequence, in the format (label, (span_start, span_end)).
     """
     spans = []
-    k = 0
-    while k < len(tag_sequence):
-        label = tag_sequence[k]
+    classes_to_ignore = classes_to_ignore or []
+    index = 0
+    while index < len(tag_sequence):
+        label = tag_sequence[index]
         if label[0] == 'U':
-            spans.append((label.partition('-')[2], (k, k)))
+            spans.append((label.partition('-')[2], (index, index)))
         elif label[0] == 'B':
-            start = k
+            start = index
             while label[0] != 'L':
-                k += 1
-                if k >= len(tag_sequence):
+                index += 1
+                if index >= len(tag_sequence):
                     raise InvalidTagSequence
-                label = tag_sequence[k]
+                label = tag_sequence[index]
                 if not (label[0] == 'I' or label[0] == 'L'):
                     raise InvalidTagSequence
-            spans.append((label.partition('-')[2], (start, k)))
+            spans.append((label.partition('-')[2], (start, index)))
         else:
             if label != 'O':
                 raise InvalidTagSequence
-        k += 1
-    return spans
+        index += 1
+    return [span for span in spans if span[0] not in classes_to_ignore]
```

Comment:
not for this PR, but it looks like this logic allows for invalid tag sequences like

B-PER -> I-ORG -> L-MISC

?

we should probably open an issue for that

Reply:
Good catch, in this case it'd mark the entire span as `MISC`.